users.js

```diff
@@ -1273,10 +1348,17 @@ User = (function () {
 		}
 
 		for (var ip in this.ips) {
-			lockedIps[ip] = userid;
+			lockedIps[ip] = {
+				time: time,
+				userid: userid
+			};
 		}
-		if (this.autoconfirmed) lockedUsers[this.autoconfirmed] = this.userid;
-		if (this.authenticated) lockedUsers[this.userid] = this.userid;
+		var entry = {
+			time: time,
+			userid: this.userid
+		};
+		if (this.autoconfirmed) lockedUsers[this.autoconfirmed] = entry;
+		if (this.authenticated) lockedUsers[this.userid] = entry;
```

Comment:
Using the pointer to the same entry? How risky is this?

I suppose the entry is basically constant, so not too much.